doc/specs/#147 - Support different release channels [released, beta, alpha].md

```diff
@@ -0,0 +1,259 @@
+  | Google.Chrome.installer.Stable.yaml
+  | Google.Chrome.locale.en-US.yaml
+  | Google.Chrome.locale.nb-NO.yaml
+  | Google.Chrome.yaml
+/manifests/g/Google/Chrome/2.0
+  | Google.Chrome.installer.canary.yaml
+  | Google.Crome.installer.dev.yaml
+  | Google.Chrome.locale.en-US.yaml
+  | Google.Chrome.locale.nb-NO.yaml
+  | Google.Chrome.yaml
+```
+
+To maintain backwards compatability, `null` should be treated as a valid `DefaultChannel`. If the default channel is not specified, `<PackageIdentifier>.installer.yaml` would be treated as the default. Additional channels may still be added to the package, and all functionality would still be present.
+
+An example of the manifest file structure when `DefaultChannel` is unset -
+```raw
+/manifests/7/7zip/7zip/1.0
+  | 7zip.7zip.installer.yaml
+  | 7zip.7zip.installer.Alpha.yaml
+  | 7zip.7zip.locale.en-US.yaml
+  | 7zip.7zip.yaml
+/manifests/7/7zip/7zip/2.0
+  | 7zip.7zip.installer.Alpha.yaml
+  | 7zip.7zip.locale.en-US.yaml
+  | 7zip.7zip.yaml
+```
+
+### CLI Behavior
+Where not specified by the user, the default release channel should be used/shown first or exclusively, then other channels in ASCII sort order. If the package manifest has a default channel specified and no versions of that package have an installer manifest matching the default channel, then the next available channel should be used/shown. Channels should be treated as *case-insensitive*.
+
+The channel `Default` shall be treated as a reserved token. When used in commands, default should refer to the default channel specified in the version manifest. In command output, when the default channel is named, the name should be shown. If the default channel is null, then `Default` should be shown. Manifest validation should throw an error when `DefaultChannel` is `Default`
+
+#### winget search
+In order to avoid package duplication in search results, only the default channel should be shown unless otherwise specified. Users should be able to search for packages by release channel in addition to all other search filters.
+
+Some valid searches:
+```powershell
+PS> winget search --channel <channel> # Return all packages which have a channel matching <channel>
+PS> winget search --channel "" # Return all packages which have a null channel
+PS> winget search <query> --channel <channel> # Return packages matching <query> which have a channel matching <channel>
+PS> winget search --tag <tag> --channel <channel> # Return packages with a tag matching <tag> which have a channel matching <channel>
+...
+```
+
+#### winget show
+When the user specifies `--versions`, the versions for all channels along with the channels should be shown. The versions should be grouped by channel, sorted by channel, then sorted by version descending
+When the user specifies `--channels`, the available channels for the package should be shown. If a package contains a named default channel and unnamed (null) channels, only the named channels should be shown
```

Comment:
I'm not following this part, but I assert that there should be only one empty channel, and any number of non-empty channels.

Reply:
I think this goes back to your point about whether or not a named channel could or should be allowed to be default